llvm-general/test/LLVM/General/Test/Linking.hs

```diff
@@ -59,9 +59,9 @@ tests = testGroup "Linking" [
         ]      
 
     Module { moduleDefinitions = defs } <- withContext $ \context -> 
-      withModuleFromAST' context ast0 $ \m0 ->
-        withModuleFromAST' context ast1 $ \m1 -> do
-          runExceptT $ linkModules False m0 m1
-          moduleAST m0
+      withModuleFromAST' context ast0 $ \dest -> do
+        src <- failInIO $ astToFFIModule context ast1
+        failInIO $ linkModules dest src
```

Comment:
Agreed it'd be a pain to handle statically. Can we handle it dynamically? The old version used to cannibalize the module anyway, so it was effectively mutated. How about an approach where the Link function mutates the Module, so the dispose doesn't crash? I.e. change Module from
Module (Ptr FFI.Module) to Module (IORef (Ptr FFI.Module)) and set it to null in the link function, or something like that.